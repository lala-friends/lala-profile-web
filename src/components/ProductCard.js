import React, {Component} from 'react';
import PropTypes from 'prop-types';

class ProductCard extends Component {
    moveToProductDetail = () => {
        this.props.history.push(`/product/${this.props.name}`);
    }
    render() {
        return (
            <div className='card' onClick={this.moveToProductDetail}> 
                <div className='image-wrapper'>
                    <img id='imageUrl' alt='제품 컨셉 이미지' className='square-image' src={this.props.imageUrl} />
                </div>

                <div id='productName' className='item-title'>{this.props.name}</div>
                <div className={`group ${!this.props.introduce&&'inactive'}`}>
                    <div className='fieldName'>한줄소개</div>
                    <div id='productSummary'>{this.props.introduce}</div>
                </div>
                <div className={`group ${!this.props.techs&&'inactive'}`}>
                    <div className='fieldName'>기술 스택</div>
                    <div id='techs'>
                    {
                        this.props.techs && this.props.techs.map((tech, key) => (
                            <div className="badge badge-pill badge-info" key={key}>{tech}</div>
                        ))
                    }
                    </div>
                </div>
                <div className={`group ${!this.props.role && 'inactive'}`}>
                    <div className='fieldName'>역할</div>
                    <div id='role'>{this.props.role}</div>
                </div>

            </div>
        )
    }
}

ProductCard.propTypes = {
    imageUrl: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    introduce: PropTypes.string,
    techs: PropTypes.array,
    role: PropTypes.string
}

export default ProductCard;