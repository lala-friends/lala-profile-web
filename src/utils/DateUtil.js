export const dateYYYYMMDD = (stringDate) => {
    const date = new Date(stringDate.toString());
    return date.getFullYear() + '-' + pad(date.getMonth()+1, 2) + '-' + pad(date.getDate(), 2);
};

export const dateYYYYMMDDHHMMSS = (stringDate) => {
    const date = new Date(stringDate.toString());
    return date.getFullYear() + '-' + pad(date.getMonth()+1, 2) + '-' + pad(date.getDate(), 2) + ' ' +
        pad(date.getHours(), 2) + ':' + pad(date.getMinutes(), 2) + ':' + pad(date.getSeconds(), 2);
}

const pad = (number, width) => {
    let numberStr = number.toString();
    if (numberStr.length >= width) {
        return numberStr;
    } else {
        for(let i=width-numberStr.length; i>0; i--)
            numberStr = '0' + numberStr;
        return numberStr;
    }
};