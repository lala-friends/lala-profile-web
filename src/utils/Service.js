import axios from 'axios';
import {baseURL} from '../config';

class Service {
    constructor() {
        const service = axios.create({
            baseURL: baseURL
        });
        this.service = service;
    }

    get(path, callback) {
        return this.service.get(path).then(
            (response) => callback(response.status, response.data)
        );
    }

    post(path, payload, callback) {
        return this.service.post(path, payload).then(
            (response) => callback(response.status, response.data)
        );
    }

    put(path, payload, callback) {
        return this.service.put(path, payload).then(
            (response) => callback(response.status, response.data)
        );
    }

    delete(path, callback) {
        return this.service.delete(path).then(
            (response) => callback(response.status, response.data)
        );
    }
}

export default Service;