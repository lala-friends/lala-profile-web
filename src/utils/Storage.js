import firebase from 'firebase';
import {firebaseConfig} from '../config';
firebase.initializeApp(firebaseConfig);

export const uploadImage = (file) => {
    const task = firebase.storage().ref().child(file.name).put(file, { contentType: file.type });
    return task;
}