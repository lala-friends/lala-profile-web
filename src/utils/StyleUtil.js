export const rgbToHexa = (rgb) => {
    return getHexa(...rgb.substring(4, rgb.length-1).split(', '));
}
const getHexa = (red, green, blue) => {
    const rgb = blue | (green << 8) | (red << 16);
    return '#' + (0x1000000 + rgb).toString(16).slice(1)
};

export const colorPalette = ['#ffa8a8', '#faa2c1', '#eebefa', '#d0bfff', '#bac8ff', '#a5d8ff', '#c5f6fa', '#99e9f2', '#96f2d7', '#b2f2bb', '#ffe066', '#ff922b'];