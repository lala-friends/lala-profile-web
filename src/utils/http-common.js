import axios from 'axios';
import https from 'https';
import {baseURL} from "../config";

export default axios.create({
    baseURL: baseURL,
    httpsAgent: new https.Agent({
        rejectUnauthorized: false
    }),
});
