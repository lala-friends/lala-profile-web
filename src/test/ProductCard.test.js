import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import ProductCard from '../components/ProductCard';
import {colorPalette} from '../utils/StyleUtil';

describe('ProductCard', () => {
    test('ProductCard 컴포넌트를 렌더링 한다.', () => {
        const wrapper = shallow(<ProductCard name='lala-clipping' 
                                    imageUrl='www.imageUrl.com' 
                                    introduce='간단 소개' 
                                    techs={['android']}
                                    role='개발'/>);

        expect(wrapper.find('#imageUrl').props().src).toEqual('www.imageUrl.com');
        expect(wrapper.find('#productName').text()).toEqual('lala-clipping');
        expect(wrapper.find('#productSummary').text()).toEqual('간단 소개');
        expect(wrapper.find('#techs .badge').length).toEqual(1);
        expect(wrapper.find('#techs .badge').at(0).text()).toEqual('android');
        expect(wrapper.find('#role').text()).toEqual('개발');
    });

    test('props로 넘겨지지 않은 요소는, 화면에 보여지지 않는다. ', () => {
        const wrapper = shallow(<ProductCard name='lala-clipping' 
                                            imageUrl='www.imageUrl.com' 
                        />);

        expect(wrapper.find('#techs').parent().props().className).toContain('inactive');
        expect(wrapper.find('#productSummary').parent().props().className).toContain('inactive');
        expect(wrapper.find('#role').parent().props().className).toContain('inactive');
    });

    test('프로덕트카드를 클릭하면 프로덕트 상세페이지로 이동한다.', () => {
        const push = jest.fn();
        const wrapper = shallow(<ProductCard name='lala-clipping' 
                                            imageUrl='www.imageUrl.com' 
                                            history={{push: push}}
        />);

        wrapper.find('#productCard').simulate('click');

        expect(push).toHaveBeenCalledWith('/product/lala-clipping');
    });
});