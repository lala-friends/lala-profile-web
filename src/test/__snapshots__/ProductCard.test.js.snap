// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard ProductCard 컴포넌트를 렌더링 한다. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductCard
    imageUrl="www.imageUrl.com"
    introduce="간단 소개"
    name="lala-clipping"
    role="개발"
    techs={
        Array [
            "android",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="image-wrapper"
>
          <img
                    alt="제품 컨셉 이미지"
                    className="square-image"
                    id="imageUrl"
                    src="www.imageUrl.com"
          />
</div>,
        <div
          className="item-title"
          id="productName"
>
          lala-clipping
</div>,
        <div
          className="group false"
>
          <div
                    className="fieldName"
          >
                    한줄소개
          </div>
          <div
                    id="productSummary"
          >
                    간단 소개
          </div>
</div>,
        <div
          className="group false"
>
          <div
                    className="fieldName"
          >
                    기술 스택
          </div>
          <div
                    id="techs"
          >
                    <div
                              className="badge badge-pill badge-info"
                    >
                              android
                    </div>
          </div>
</div>,
        <div
          className="group false"
>
          <div
                    className="fieldName"
          >
                    역할
          </div>
          <div
                    id="role"
          >
                    개발
          </div>
</div>,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="제품 컨셉 이미지"
            className="square-image"
            id="imageUrl"
            src="www.imageUrl.com"
/>,
          "className": "image-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "제품 컨셉 이미지",
            "className": "square-image",
            "id": "imageUrl",
            "src": "www.imageUrl.com",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "lala-clipping",
          "className": "item-title",
          "id": "productName",
        },
        "ref": null,
        "rendered": "lala-clipping",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fieldName"
>
              한줄소개
</div>,
            <div
              id="productSummary"
>
              간단 소개
</div>,
          ],
          "className": "group false",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "한줄소개",
              "className": "fieldName",
            },
            "ref": null,
            "rendered": "한줄소개",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "간단 소개",
              "id": "productSummary",
            },
            "ref": null,
            "rendered": "간단 소개",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fieldName"
>
              기술 스택
</div>,
            <div
              id="techs"
>
              <div
                            className="badge badge-pill badge-info"
              >
                            android
              </div>
</div>,
          ],
          "className": "group false",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "기술 스택",
              "className": "fieldName",
            },
            "ref": null,
            "rendered": "기술 스택",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="badge badge-pill badge-info"
>
                  android
</div>,
              ],
              "id": "techs",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": "android",
                  "className": "badge badge-pill badge-info",
                },
                "ref": null,
                "rendered": "android",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fieldName"
>
              역할
</div>,
            <div
              id="role"
>
              개발
</div>,
          ],
          "className": "group false",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "역할",
              "className": "fieldName",
            },
            "ref": null,
            "rendered": "역할",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "개발",
              "id": "role",
            },
            "ref": null,
            "rendered": "개발",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="image-wrapper"
>
            <img
                        alt="제품 컨셉 이미지"
                        className="square-image"
                        id="imageUrl"
                        src="www.imageUrl.com"
            />
</div>,
          <div
            className="item-title"
            id="productName"
>
            lala-clipping
</div>,
          <div
            className="group false"
>
            <div
                        className="fieldName"
            >
                        한줄소개
            </div>
            <div
                        id="productSummary"
            >
                        간단 소개
            </div>
</div>,
          <div
            className="group false"
>
            <div
                        className="fieldName"
            >
                        기술 스택
            </div>
            <div
                        id="techs"
            >
                        <div
                                    className="badge badge-pill badge-info"
                        >
                                    android
                        </div>
            </div>
</div>,
          <div
            className="group false"
>
            <div
                        className="fieldName"
            >
                        역할
            </div>
            <div
                        id="role"
            >
                        개발
            </div>
</div>,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="제품 컨셉 이미지"
              className="square-image"
              id="imageUrl"
              src="www.imageUrl.com"
/>,
            "className": "image-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "제품 컨셉 이미지",
              "className": "square-image",
              "id": "imageUrl",
              "src": "www.imageUrl.com",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "lala-clipping",
            "className": "item-title",
            "id": "productName",
          },
          "ref": null,
          "rendered": "lala-clipping",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="fieldName"
>
                한줄소개
</div>,
              <div
                id="productSummary"
>
                간단 소개
</div>,
            ],
            "className": "group false",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "한줄소개",
                "className": "fieldName",
              },
              "ref": null,
              "rendered": "한줄소개",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "간단 소개",
                "id": "productSummary",
              },
              "ref": null,
              "rendered": "간단 소개",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="fieldName"
>
                기술 스택
</div>,
              <div
                id="techs"
>
                <div
                                className="badge badge-pill badge-info"
                >
                                android
                </div>
</div>,
            ],
            "className": "group false",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "기술 스택",
                "className": "fieldName",
              },
              "ref": null,
              "rendered": "기술 스택",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="badge badge-pill badge-info"
>
                    android
</div>,
                ],
                "id": "techs",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "android",
                    "className": "badge badge-pill badge-info",
                  },
                  "ref": null,
                  "rendered": "android",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="fieldName"
>
                역할
</div>,
              <div
                id="role"
>
                개발
</div>,
            ],
            "className": "group false",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "역할",
                "className": "fieldName",
              },
              "ref": null,
              "rendered": "역할",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "개발",
                "id": "role",
              },
              "ref": null,
              "rendered": "개발",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
